pkgpanda/__init__.py

```diff
@@ -678,9 +712,22 @@ def symlink_all(src, dest):
 
         active_buildinfo_full = {}
 
+        dcos_service_configuration = self._get_dcos_configuration_template()
+
         # Building up the set of users
         sysusers = UserManagement(self.__manage_users, self.__add_users)
 
+        def _get_service_names(_dir):
+            service_files = list(map(os.path.basename,
+                                     glob.glob(_dir + "/dcos.target.wants/*.service") +
+                                     glob.glob(_dir + "/dcos.target.wants_master/*.service") +
+                                     glob.glob(_dir + "/dcos.target.wants_slave/*.service") +
+                                     glob.glob(_dir + "/dcos.target.wants_slave_public/*.service")))
+            if not service_files:
```

Comment:
So, the design this. If there is a **sysctl configuration** that is given without a service file, we want to fail early.

1) Letting it go with empty list in Line 726 will not work as it will cause Exception in line 729.
2) We can send empty `[]` that fail early. Just a choice. I am okay to change.

Btw, the way of returning None or list is **very common**  if you use `Optional` in other programming languages. Here we do it by checking for None in the calling code and if not using it as List in the caller.